scripts/g.extension/g.extension.py

```diff
@@ -256,15 +261,21 @@ def __init__(
 
     def _get_version(self):
         """Get the installed git version"""
-        git_version = gs.Popen(["git", "--version"], stdout=PIPE)
-        return float(
-            ".".join(
-                gs.decode(git_version.communicate()[0])
-                .rstrip()
-                .rsplit(" ", 1)[-1]
-                .split(".")[0:2]
+        git_version = gs.Popen([self._git, "--version"], stdout=PIPE, stderr=PIPE)
+        git_version, stderr = git_version.communicate()
+        if stderr:
+            gs.fatal(
+                _("Failed to get Git version.\n{error}").format(
+                    gs.decode(stderr),
+                )
             )
-        )
+        git_version = re.search(r"\d+.(\d+.\d+|\d+)", gs.decode(git_version))
+        if not git_version:
+            gs.fatal(_("Failed to get Git version."))
+        git_version = git_version.group()
+        if git_version.count(".") == 2:
```

Comment:
Unfortunately, I cannot test this weekend. Using re looks like a good choice in general. Is there any chance that count(".") != 2?

Reply:
Regular expression `git_version = re.search(r"\d+.(\d+.\d+|\d+)", gs.decode(git_version))` catch following Git version example:

1. major.minor.patch -> 2.39.2
2.  major.minor -> 2.39

Condition `if git_version.count(".") == 2:` handle first case (line 277 extract major.minor string from major.minor.patch string 2.39.2 -> 2.39). 

https://github.com/OSGeo/grass/blob/b5a119a90050f6f79a3d0fc4e332bc3db897c581/scripts/g.extension/g.extension.py#L276-L277

if `git_version` var string value is **major.minor** string e.g. 2.39, conversion to float is done on the line 278.

https://github.com/OSGeo/grass/blob/b5a119a90050f6f79a3d0fc4e332bc3db897c581/scripts/g.extension/g.extension.py#L278

Reply:
```python
import re

In [7]: re.search(r"\d+.(\d+.\d+|\d+)", "git version 2.39.2 (Apple Git-143)").group()
Out[7]: '2.39.2'

In [8]: re.search(r"\d+.(\d+.\d+|\d+)", "git version 2.39 (Apple Git-143)").group()
Out[8]: '2.39'
```